docs/workloads/azure/data/components/infrastructure/core_data_platform_azure.md

```diff
@@ -10,28 +10,84 @@ keywords:
-Deployment details (usage manual, CLI commands) should be captured in
-[getting_started](../../getting_started/core_data_platform_deployment_azure.md).
+Core data platform resources are defined within Terraform templates and grouped inside
+[deploy/azure](https://github.com/amido/stacks-azure-data/tree/main/deploy/azure) directory.
+There are two subfolders in this directory:
 
-This should be an extension to [intro_data_azure.md](../../intro_data_azure.md).
+* `infra`
+* `networking`
 
-### Infrastructure deployed
+## Infra
 
-* Resource Group
-* Azure Data Factory
-* Azure Data Lake Storage Gen2
-* Azure Blob Storage (for config files)
-* Azure Key Vault
-* Log Analytics Workspace
-* Databricks Workspace
-    * Azure Key Vault-backed secret scope
+`infra` subfolder contains the following definitions:
 
+1. **Resource Group**
+2. **Azure SQL Database** sample instance with database schemas
+3. **Key Vault**
+4. **Key Vault Secrets**:
+    * Connection strings for the created SQL databases
+    * Passwords to the SQL Databases
+    * Service Principal Secret
+    * Azure Tenant ID – Directory ID for Azure Active Directory application
+    * Azure Client ID – Application ID for Azure Active Directory application
+    * Names of the ADLS and Blob storage accounts
+    * Other secret names with empty values to be replaced manually. Existing secrets are not
+    overwritten.
+5. **Azure Data Lake Storage Gen2**
+6. **Azure Blob Storage**
+7. **Databricks Workspace** including:
+    * **Key Vault-backed secret scope**
+8. **Azure Data Factory** including:
+    * **Managed identity** for the service instance;
+    * **Managed virtual network** enabled by default. Creating an integration runtime within
+    a managed virtual network ensures the data integration process is isolated and secure.
+9. **Managed private endpoints** created in the Data Factory managed virtual network, these
+   establish private links to Azure resources, such as:
+    * Blob Storage
+    * Azure Data Lake Storage
+    * Key Vault
+    * SQL Database
+10. **Role assignments** that assign ADF managed identity roles to access the resources linked by
+   the private endpoints, as well as the Databricks workspace.
+11. **Log Analytics Workspace**
 
-& Networking, VNet, any additional security info
+## Networking
+
+Using a private network is the default behaviour in Stacks Azure Data Platform. `networking`
+subfolder contains configurations for the created network and subnetworks, at its core using
+[azurerm-hub-spoke](https://github.com/amido/stacks-terraform/tree/master/azurerm/modules/azurerm-hub-spoke)
+Stacks Terraform module.
+
+The following diagram shows network configuration in all three environments:
+
+* dev (`is_hub: true`)
+* nonprod (`is_hub: false`)
+* prod (`is_hub: false`)
+
+![network_hub_spoke.png](../../images/network_hub_spoke.png)
+
+### Databricks secure cluster connectivity
```

Comment:
@Trishisingh here is the part about Databricks secure connectivity - do you want me to add more info here?